pallets/collator-assignment/src/tests/assign_full.rs

```diff
@@ -19,9 +19,22 @@ use {
         assignment::{Assignment, AssignmentError},
         tests::Test,
     },
+    rand::{seq::SliceRandom, SeedableRng},
+    rand_chacha::ChaCha20Rng,
     sp_std::collections::btree_map::BTreeMap,
 };
 
+#[allow(unused_assignments)]
+fn no_shuffle() -> Option<impl FnOnce(&mut Vec<u64>)> {
+    let mut shuffle = Some(move |_collators: &mut Vec<u64>| {
+        // Create an empty closure to allow the compiler to infer the return type
+    });
+    // But don't return that closure, instead return `None`
+    shuffle = None;
```

Comment:
I feel that I am doing something wrong here, but I can't just `return None::<_>` because the compiler doesn't know the type of `_`.

Reply:
Improved this, thanks @nanocryk